src/ORM/Search/SearchContextForm.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+namespace SilverStripe\ORM\Search;
+
+use SilverStripe\Control\RequestHandler;
+use SilverStripe\Core\ClassInfo;
+use SilverStripe\Forms\CompositeField;
+use SilverStripe\Forms\FieldList;
+use SilverStripe\Forms\Form;
+use SilverStripe\Forms\GridField\GridField;
+use SilverStripe\Forms\GridField\GridField_FormAction;
+use SilverStripe\Forms\GridField\GridFieldFilterHeader;
+use SilverStripe\ORM\FieldType\DBHTMLText;
+use SilverStripe\View\SSViewer;
+
+/**
+ * A form for using a search context in the CMS.
+ *
+ * Note that the form submission goes through to the index action of the controller by default,
+ * and is intended to return the same markup you would ordinarily get for that controller but with
+ * the results filtered appropriately.
+ */
+class SearchContextForm extends Form
+{
+    private SearchContext $searchContext;
+
+    private ?string $searchField = null;
+
+    private static $casting = [
+        'SchemaData' => 'Text',
+        'getSchemaData' => 'Text',
+    ];
+
+    public function __construct(RequestHandler $controller, SearchContext $searchContext, $name = 'SearchForm')
+    {
+        $this->searchContext = $searchContext;
+        $searchFields = clone $searchContext->getSearchFields();
+        $this->setDefaultSearchField($searchFields, $controller);
+        $this->prepareSearchFieldsForCMS($searchFields);
+        parent::__construct($controller, $name, $searchFields);
+        $this->setHTMLID(ClassInfo::shortName($controller) . '_' . $name);
+        $this->disableSecurityToken();
```

Comment:
Why are we disabling this?

Reply:
Because we always were

- https://github.com/silverstripe/silverstripe-framework/blob/5f908da2aae2d209b94a476cc0c3d6f4b57c3abd/src/Forms/GridField/GridFieldFilterHeader.php#L421
- https://github.com/silverstripe/silverstripe-cms/blob/c989e4f39bc5fad3527875ae3fce1502dfdcec1b/code/Controllers/CMSMain.php#L1123

Reply:
Cool, could you add that comment back in above it so others know why it's done

Reply:
Cool, could you add that comment back in above it so others know why it's done

Reply:
Done